pandas/tests/extension/base/getitem.py

```diff
@@ -330,13 +341,31 @@ def test_take_non_na_fill_value(self, data_missing):
         self.assert_extension_array_equal(result, expected)
 
     def test_take_pandas_style_negative_raises(self, data, na_value):
-        with pytest.raises(ValueError):
+        with pytest.raises(ValueError, match=""):
             data.take([0, -2], fill_value=na_value, allow_fill=True)
 
     @pytest.mark.parametrize("allow_fill", [True, False])
     def test_take_out_of_bounds_raises(self, data, allow_fill):
         arr = data[:3]
-        with pytest.raises(IndexError):
+        if (arr.dtype.name == "arrow_string") | ("Sparse" in arr.dtype.name):
+            msg = "out of bounds value in 'indices'."
+        elif arr.dtype.name == "json":
+            msg = (
+                "Index is out of bounds or cannot do a non-empty take "
+                "from an empty array."
+            )
+        else:
+            if allow_fill:
+                msg = "indices are out-of-bounds"
+            else:
+                if ("numpy" not in str(type(arr))) | (
+                    arr.dtype.name not in ["string"]
+                ):
+                    msg = "index 3 is out of bounds for axis 0 with size 3"
+                else:
+                    msg = "index 3 is out of bounds for size 3"
+
+        with pytest.raises(IndexError, match=msg):
```

Comment:
I don't think we need to match the whole message here, something like
```python
        with pytest.raises(IndexError, match='out of bounds|out-of-bounds'):
            arr.take(np.asarray([0, 3]), allow_fill=allow_fill)
```
should be enough